contracts/vesting/src/contract.rs

```diff
@@ -0,0 +1,636 @@
+
+        verify_vesting(&env, &from, amount, &token_client)?;
+        token_client.transfer(&from, &to, &amount);
+
+        env.events().publish(
+            (
+                "Transfer vesting",
+                "Transfering vesting between accounts: from: {}, to: {}, amount: {}",
+            ),
+            (from, to, amount),
+        );
+
+        Ok(())
+    }
+
+    fn burn(env: Env, sender: Address, amount: i128) -> Result<(), ContractError> {
+        sender.require_auth();
+
+        if amount <= 0 {
+            log!(&env, "Vesting: Burn: Invalid burn amount");
+            panic_with_error!(env, ContractError::InvalidBurnAmount);
+        }
+
+        let remainder = get_vesting_total_supply(&env) - amount;
+        match remainder < 0 {
+            true => {
+                log!(
+                    &env,
+                    "Vesting: Burn: Critical error - total supply cannot be negative"
+                );
+                panic_with_error!(env, ContractError::Std);
+            }
+            false => update_vesting_total_supply(&env, remainder),
+        };
+        let token_client = token_contract::Client::new(&env, &get_token_info(&env).address);
+
+        verify_vesting(&env, &sender, amount, &token_client)?;
+        token_client.burn(&sender, &amount);
+
+        env.events().publish(("Burn", "Burned tokens: "), amount);
+
+        Ok(())
+    }
+
+    fn mint(env: Env, sender: Address, to: Address, amount: i128) {
+        sender.require_auth();
+
+        if amount <= 0 {
+            log!(&env, "Vesting: Mint: Invalid mint amount");
+            panic_with_error!(env, ContractError::InvalidMintAmount);
+        }
+
+        // check if sender is minter
+        if sender != get_minter(&env).address {
+            log!(&env, "Vesting: Mint: Not authorized to mint");
+            panic_with_error!(env, ContractError::NotAuthorized);
+        }
+
+        // update supply and capacity
+        let updated_total_supply = get_vesting_total_supply(&env)
+            .checked_add(amount)
+            .unwrap_or_else(|| {
+                log!(
+                    &env,
+                    "Vesting: Mint: Critical error - total supply overflow"
+                );
+                panic_with_error!(env, ContractError::Std);
+            });
+
+        update_vesting_total_supply(&env, updated_total_supply);
+
+        let limit = get_minter(&env).capacity.value(env.ledger().timestamp());
+        if updated_total_supply >= limit as i128 {
+            log!(&env, "Vesting: Mint: total supply over the capacity");
+            panic_with_error!(env, ContractError::SupplyOverTheCap);
+        }
+
+        // mint to recipient
+        let token_client = token_contract::Client::new(&env, &get_token_info(&env).address);
+        token_client.mint(&to, &amount);
+
+        env.events().publish(("Mint", "Minted tokens: "), amount);
```

Comment:
Modify this in other events as well.

```suggestion
        env.events().publish(("Mint", "Minted tokens: "), amount);
        env.events().publish(("Mint", "Recipient: "), to);
```